src/vault/authenticate_vault/kubernetes.rs

```diff
@@ -0,0 +1,58 @@
+use std::fs::File;
+use std::io::Read;
+use crate::errors::error::VaultError;
+use crate::vault::vault_config::VaultConfig;
+use crate::vault::vault_service::VaultService;
+use vaultrs::{
+    api::AuthInfo,
+    auth::kubernetes,
+    client::{Client, VaultClient, VaultClientSettingsBuilder},
+};
+use async_trait::async_trait;
+use crate::vault::authenticate_vault::authenticate_vault_trait::AuthenticateVault;
+
+pub struct AuthenticateKubernetesVault;
+
+#[async_trait]
+impl AuthenticateVault for AuthenticateKubernetesVault {
+    fn get_vault_settings(&self, config: VaultConfig) -> VaultClientSettingsBuilder {
+        return VaultClientSettingsBuilder::default()
+            .address(config.address.clone())
+            .timeout(Some(config.client_timeout)).clone();
+    }
+
+    fn get_jwt_token(&self, config: VaultConfig) -> Result<Option<String>, VaultError> {
+        let mut token = String::new();
+        let mut file = match File::open(config.clone().token_path.unwrap().as_str()) {
+            Ok(res) => res,
+            Err(err) => {
+                return Err(VaultError::new(format!("File does not exists in given path: {}\n{}", config.token_path.clone().unwrap(), err)));
```

Comment:
This error description is wrong. How do you know the file does not exists? Maybe there was a permission issue? Why not just `return Err(err)`?